install.ps1

```diff
@@ -1,98 +1,389 @@
+Please provide a valid Git commit hash.
 ---------------------------------------------------------------------------
 "@
-        exit
-    } else {
-@"
+        exit 1
+    }
+}
 
-Thank You for downloading Dagger!
+function Get-DownloadUrl {
+    $fileName = Get-FileName
 
------------------------------------------------------
-Dagger has been saved at <YOUR HOME FOLDER>/dagger/
-Please add dagger.exe to your PATH in order to use it
-----------------------------------------------------
+    if (-not [string]::IsNullOrWhiteSpace($DaggerCommit)) {
+        return "https://dl.dagger.io/dagger/main/${DaggerCommit}/${fileName}"
+    }
 
-"@
+    return "https://dl.dagger.io/dagger/releases/${DaggerVersion}/${fileName}"
+}
+
+# Used for interactive mode to get a true or false response from the user
+function Get-TrueFalse {
+    Param (
+        [Parameter(Mandatory = $true)][string]$Message,
+        [Parameter(Mandatory = $true)][bool]$Default
+    )
+
+    $response = ""
+    while ($response -notmatch "^(y|n)$") {
+        $response = Read-Host -Prompt $Message
+        if ([string]::IsNullOrWhiteSpace($response)) {
+            $response = $Default ? "y" : "n"
+            break
+        }
     }
 
+    return $response.StartsWith("y")
 }
 
-function latest_version {
-    $response = Invoke-RestMethod 'http://dl.dagger.io/dagger/latest_version' -Method 'GET' -Body $body -ErrorAction SilentlyContinue -ErrorVariable DownloadError
-    If ($DownloadError) {
-@"
+function Find-LatestVersion {
+    try {
+        $body = $null
+        $response = Invoke-RestMethod "https://dl.dagger.io/dagger/latest_version" -Body $body -ErrorVariable LatestVersionError
+        
+        if ($LatestVersionError) {
+            Write-Host 
+            @"
 ---------------------------------------------------------------------------
 Houston we have a problem!
-
-Apparently we had an issue in downloading the file, please try again
-run the script and if it still fail please open an issue on the Dagger repo.
+Apparently we had an issue finding the latest version of Dagger.
+Please check https://docs.dagger.io/install
 ----------------------------------------------------------------------------
 "@
-        exit
+            exit 1
+        }
+
+        $latestVersion = $response -replace "v", ""
+        Write-Host "Latest version of Dagger is v$latestVersion"
+
+        return [System.Management.Automation.SemanticVersion]::Parse($latestVersion)
+    }
+    catch {
+        if ($_.Exception.Response.StatusCode -eq 'Found') {
+            $uri = $_.Exception.Response.Headers.Location
+            $latestVersionPart = $uri.ToString().Split('/') | Where-Object { $_ -match '^v\d+' }
+            $latestVersion = $latestVersionPart -replace "v", ""
+            Write-Host "Latest version of Dagger is v$latestVersion"
+
+            return [System.Management.Automation.SemanticVersion]::Parse($latestVersion)
+        }
+        else {
+            Write-Host @"
+
+"@
```

Comment:
:eyes: should there be something here?

Reply:
Uhhh, i think my eyes were missing something on the previous rebase + merge. :D